File: solutions/javascript/atbash-cipher/1/atbash-cipher.js
//
// This is only a SKELETON file for the 'Atbash Cipher' exercise. It's been provided as a
// convenience to get you started writing code faster.
//
const isAlpha = (char) => /^[a-z]$/.test(char);
const isDigit = (char) => /^[0-9]$/.test(char);

export const encode = (text) => {
  let t = text.toLowerCase();
  let encoded = '';

  for (let i = 0; i < t.length; i++) {
    if (isAlpha(t[i])) {
      encoded += String.fromCharCode('z'.charCodeAt(0) - (t.charCodeAt(i) - 97));
    } else if (isDigit(t[i])) {
      encoded += t[i];
    }
    // ignore punctuation and spaces
  }

  const cleaned = encoded.replace(/\s+/g, '');
  let grouped = '';

  for (let i = 0; i < cleaned.length; i += 5) {
    grouped += cleaned.slice(i, i + 5) + ' ';
  }

  return grouped.trim();
};


export const decode = (ciphertext) => {
 let t = ciphertext.toLowerCase();
  let decoded = '';

  for (let i = 0; i < t.length; i++) {
    if (isAlpha(t[i])) {
      decoded += String.fromCharCode('z'.charCodeAt(0) - (t.charCodeAt(i) - 97));
    } else if (isDigit(t[i])) {
      decoded += t[i];
    }
    
  }
  return decoded
};

File: solutions/python/atbash-cipher/1/atbash_cipher.py
def encode(plain_text):


  
    plain_text = plain_text.lower()
    
    encoded = ""


    for ch in plain_text:
        
       
        if 'a' <= ch <= 'z':
            mirrored = chr(ord('z') - (ord(ch) - ord('a')))
            encoded += mirrored
        
    
        elif ch.isdigit():
            encoded += ch
        

        else:
            continue


    grouped = ""
    
    for i in range(0, len(encoded), 5):
        grouped += encoded[i:i+5] + " "

    return grouped.strip()


def decode(ciphered_text):
    

    
    ciphered_text = ciphered_text.replace(" ", "")

    decoded = ""

    for ch in ciphered_text:

        if 'a' <= ch <= 'z':
            mirrored = chr(ord('z') - (ord(ch) - ord('a')))
            decoded += mirrored

        elif ch.isdigit():
            decoded += ch

    return decoded

